budgie-hotcorners/src/budgie-hotcorners.vala

```diff
@@ -0,0 +1,438 @@
+                stays the same, go for it!
+                */
+
+                int new_area = get_area(x, y);;
+                if (new_area != curr_area &&
+                    !skip_action &&
+                    !screensaver_runs
+                ) {
+                    /*
+                    if we're not on the same spot any more, decide what  to do
+                    */
+                    if (
+                        curr_area == ActionArea.NONE &&
+                        new_area !=  ActionArea.NONE
+                    ) {
+                        delayed_action(new_area, xses, yses);
+                    }
+                    else if (
+                        curr_area != ActionArea.NONE &&
+                        new_area != ActionArea.NONE
+                    ) {
+                        if (!fired) {
+                            delayed_action(new_area, xses, yses);
+                        }
+                    }
+                    else if (new_area == ActionArea.NONE) {
+                        fired = false;
+                    }
+                    curr_area = new_area;
+                }
+                return true;
+            });
+        }
+
+        private void get_screensaver_active () {
+            try {
+                screensaver_runs = screensaverclient.GetActive();
+            }
+            catch (Error e) {
+                debug("Failed to get information about screensaver running");
+                screensaver_runs = false;
+            }
+        }
+
+        private bool service_ison (string id) {
+            string[] names = get_dbus_namelist();
+            for (int i=0; i<names.length; i++) {
+                if (id == names[i]) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private string[] get_dbus_namelist () {
+            try {
+                return freed_client.ListNames();
+            }
+            catch (Error e) {
+                debug("Failed to get list of dbus services");
+                return {};
+            }
+        }
+
+        private void get_screensaver_client () {
+            try {
+                screensaverclient = Bus.get_proxy_sync (
+                    BusType.SESSION, "org.gnome.ScreenSaver",
+                    ("/org/gnome/ScreenSaver")
+                );
+            }
+            catch (Error e) {
+                debug("Failed to setup client for org.gnome.ScreenSaver");
+            }
+        }
+
+        private bool get_mousedown () {
+            if (!service_ison("org.UbuntuBudgie.ShufflerInfoDaemon")) {
+                return false;
+            }
+            try {
+                return shufflerinfoclient.GetMouseIsdown(1);
+            }
+            catch (Error e) {
+                debug("Failed to get mouse state from org.UbuntuBudgie.ShufflerInfoDaemon");
+                stderr.printf ("%s\n", e.message);
```

Comment:
do we need the stderr as well as the debug?  we can use %s, e.message format with debug as well.

Reply:
Ah, never edit code while cooking diner. Sure debug is enough.